src/modules/clustering/process.cpp

```diff
@@ -0,0 +1,485 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2025 Team Dissolve and contributors
+
+#include "analyser/typeDefs.h"
+#include "base/sysFunc.h"
+#include "data/elements.h"
+#include "generator/box.h"
+#include "generator/copy.h"
+#include "math/regression.h"
+#include "modules/clustering/clustering.h"
+
+bool ClusteringModule::setUp(ModuleContext &moduleContext, Flags<KeywordBase::KeywordSignal> actionSignals)
+{
+    // Check user definitions
+    if (!(a_ && b_ && (cutoff_ > 0)))
+        Messenger::error("Cluster definition invalid! Set both sites and a positive cutoff.");
+
+    // If we have strict bonding, we need to check and determine index map for hydroxyl group
+    if (directional_)
+    {
+        directionIndexes_.clear();
+        for (const auto &s : {a_, b_})
+        {
+            if (s->type() != SpeciesSite::SiteType::Fragment)
+            {
+                auto instances = s->instances();
+                for (const auto &instance : instances)
+                {
+                    if (instance.originIndices().size() != 1)
+                    {
+                        Messenger::error("Static and dynamic sites for directional bonding must be based on a single origin "
+                                         "atom (bonded to at least one hydrogen)!");
+                        return false;
+                    }
+
+                    // Find the hydroxyl hydrogens and add index to the map
+                    auto &origin = s->parent()->atom(instance.originIndices()[0]);
+                    for (const auto &bond : origin.bonds())
+                        for (const auto &atom : bond.get().atoms())
+                        {
+                            if (!atom)
+                            {
+                                Messenger::error("Inaccessible bond partner found, skipping...");
+                                continue;
+                            }
+                            if (atom->Z() == Elements::H)
+                                directionIndexes_[s].emplace(atom->index());
+                        }
+                }
+            }
+            else if (s->type() == SpeciesSite::SiteType::Fragment)
+            {
+                // For a fragment site, we're relying on "#origin, -H(#h)" or similar, with hydrogens (or whatever desired in
+                // theory should work) tagged as h If the number of hydrogens exceeds one, the definition should reflect that
+                // else only one will be calculated i.e. -H(n=2,#h)
+                auto instances = s->instances();
+                for (const auto &instance : instances)
+                {
+                    if (instance.originIndices().size() != 1)
+                    {
+                        Messenger::error("NETA defined sites for directional clustering must have a single origin atom!");
+                        return false;
+                    }
```

Comment:
`Messenger::error` returns `false` so you can do this in one line:
```suggestion
                    {
                        return Messenger::error("NETA defined sites for directional clustering must have a single origin atom!");
                    }
```